### PR TITLE
[#64] Retrieving Server's Information

### PR DESCRIPTION
**Pull request by resyfer**

### CODE REVIEW

File: src/include/internal.h [hunk 1]
```diff
@@ -264,6 +264,11 @@ extern "C" {
    "        type: counter\n" \
    "        description: pg_stat_database_conflicts_confl_deadlock"
 
+/**
+ * SQL Query for retrieving floor of server's PostgreSQL version.
+ */
+#define VERSION_SQL "SELECT split_part(split_part(version(), ' ', 2), '.', 1);"
+
```
Comment (jesperpedersen) [inline src/include/internal.h:271]: I think we can put this SQL directly in the function

File: src/include/pgexporter.h [hunk 2]
```diff
@@ -183,6 +185,8 @@ extern void* shmem;
  */
 extern void* prometheus_cache_shmem;
 
+typedef unsigned int version_t;
+
```
Comment (jesperpedersen) [inline src/include/pgexporter.h:189]: I don't think we need a `typedef` for this

File: src/include/pgexporter.h [hunk 3]
```diff
@@ -198,6 +202,7 @@ struct server
    bool new;                           /**< Is the connection new */
    bool extension;                     /**< Is the pgexporter_ext extension installed */
    int state;                          /**< The state of the server */
+   version_t version;
```
Comment (jesperpedersen) [inline src/include/pgexporter.h:205]: `char` should be enough

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

File: src/include/queries.h [hunk 4]
```diff
@@ -86,6 +86,17 @@ pgexporter_query_get_functions(int server, struct query** query);
...
 pgexporter_query_execute(int server, char* sql, char* tag, struct query** query);
 
+/**
+
+ * Extracts the server's version as a char (minimum integral data type), and sets
+ * the value of `config->servers[server].version` to it.
+ *
+ * @param server The server index in config
+ * @return Server version as a char data type
+ */
```
Comment (jesperpedersen) [inline src/include/queries.h:96]: We are not really doing that though...

Thinking about it - lets just keep it as a status code - so `int`, and assign internally in the function

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

File: src/include/queries.h (hunk 4 above)
Comment (resyfer) [reply src/include/queries.h:96]: Oh I am really sorry about this. I was travelling back to my place when I made this change and pushed as soon as I got internet, and it was a lapse of judgement on my part.

Yes I am indeed assigning the version internally and the returned value is just 0 if all's ok and 1 if not.

I'll change it back immediately.

Comment (jesperpedersen) [reply src/include/queries.h:96]: np

**Comment by jesperpedersen** [comment]
Merged.

Thanks for your contribution !

**Comment by resyfer** [comment]
Thank you for the review!

**Pull request merged**